### PR TITLE
Make the library more reusable in external projects

### PR DESCRIPTION
**Pull request by dtreskunov**
* Move addEventListener DOMContentLoaded from JS to HTML
* Add ability to specify a CORS proxy
  server (e.g. https://github.com/Rob--W/cors-anywhere)
* Preserve the image name (filename or URL)
* Handle drag-and-drop events from other browser windows
* Add `graphicsWidget.getImageDataURL()`
* Make `plotDataProvider.getData()` accept an optional data series
  object
* Make `plotDataProvider.getData()` include the name of the data
  series in the output

### CODE REVIEW

**Comment by dtreskunov** [comment]
Hi, these are the changes I made to the library while working on [WebPlotDigitizer Tableau Web Data Connector](https://github.com/dtreskunov/webplotdigitizer-wdc). I hope you would find them acceptable. Thanks!

**Comment by ankitrohatgi** [comment]
Hello,

Thank you for this pull request. I just watched the video on your project website and it looks quite interesting. I do have a little concern that since I do not have access to tableau, I might not be able to maintain these pieces to stay compatible with your connector in the future.

I will go through the code in more detail this weekend.

Regards,
Ankit

**Comment by dtreskunov** [comment]
Thanks for looking! The forwards-compatibility concern is valid, but a new WebPlotDigitizer release will not break the connector because it doesn't load WPD from outside. I would simply need to keep track of when a new version is released and update the imports after making sure it still works.

File: dev.html [hunk 1]
```diff
@@ -92,6 +92,7 @@
 <script src="javascript/widgets/webcam.js"></script>
 <script src="javascript/widgets/zoom.js"></script>
 
+<script>document.addEventListener("DOMContentLoaded", wpd.initApp, true);</script>
```
Comment (ankitrohatgi) [inline dev.html]: Why was this necessary?
